webapi/src/main/scala/org/knora/webapi/messages/admin/responder/listsmessages/ListsMessagesADM.scala

```diff
@@ -370,6 +376,8 @@ abstract class ListItemGetResponseADM(listItem: ListItemADM) extends KnoraRespon
 case class ListGetResponseADM(list: ListADM) extends ListItemGetResponseADM(list) {
 
   def toJsValue: JsValue = listGetResponseADMFormat.write(this)
+
+  def format: ListGetResponseADM = this
```

Comment:
The `format` method formats all responses of type `KnoraResponseADM` before they get returned to the client so that the ontology IRIs referenced in the response are converted to the external format. You can see the method in action in the `RouteUtilADM` on line 63: `jsonResponse = knoraResponse.format.toJsValue.asJsObject`. It's the only location where `format` is called.

As I have implemented the `format` method on the `KnoraResponseADM` trait, all extensions of it inherit it. If there is nothing to do (i.e. there are no IRIs that need to be converted) the object is just returned as it is. This is what is doing the `format` method in this case where it returns `this`.

Does this answer your question?

Reply:
Thanks for the explanation. I think the method name might be a bit misleading.